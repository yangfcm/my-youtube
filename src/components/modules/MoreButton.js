import React from "react";

const MoreButton = ({ onClickMore, children }) => {
  return (
    <div className="text-center">
      <button
        className="btn btn-light"
        style={{ width: "100%" }}
        onClick={onClickMore}
      >
        <span className="text-dark">{children ? children : "More..."}</span>
      </button>
    </div>
  );
};

export default MoreButton;
