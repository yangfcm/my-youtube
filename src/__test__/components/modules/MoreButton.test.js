import React from "react";
import { shallow } from "enzyme";
import MoreButton from "components/modules/MoreButton";

describe("Test MoreButton component", () => {
  let wrapper;
  const mockOnClickMore = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <MoreButton children={"More videos"} onClickMore={mockOnClickMore} />
    );
  });
  afterEach(() => {
    wrapper.unmount();
  });

  it("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct text on the button", () => {
    expect(wrapper.find("button").text()).toContain("More videos");
  });

  it("should call onClickMore when clicking more button", () => {
    wrapper.find("button").simulate("click");
    expect(mockOnClickMore).toHaveBeenCalled();
  });
});
