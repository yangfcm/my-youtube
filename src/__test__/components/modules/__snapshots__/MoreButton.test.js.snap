// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MoreButton component should render component 1`] = `
<div
  className="text-center"
>
  <button
    className="btn btn-light"
    onClick={[MockFunction]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <span
      className="text-dark"
    >
      More videos
    </span>
  </button>
</div>
`;
